ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -376,8 +376,11 @@ let check_operation_error ~__context ~vmr ~vmgmr ~ref ~clone_suspended_vm_enable
       match op with
       | `changing_dynamic_range ->
         let metrics = Db.VM.get_metrics ~__context ~self:ref in
+        let constr =
+          "Memory limits must satisfy: \
+          static_min <= dynamic_min = dynamic_max = static_max" in
         if nested_virt ~__context ref metrics then
-          Some (Api_errors.vm_cant_use_dyn_memory, [ref_str])
+          Some (Api_errors.memory_constraint_violation, [constr])
```

Comment:
I like that because it is not obvious to me when the lifecycle tests are executed. As of now, we definitely prevent a VM from starting with an illegal memory configuration. Can the configuration be changed and become effective while the VM is running? Because this is what needs to be prevented.